### DISCUSSION TITLE
About VAO,VBO

### DISCUSSION THREAD

**Discussion by DaniaNouro** [MuhammadSulaiman001/opengl-lab#16] [Q&A]
Hello, I have many questions. 1: Suppose I want to draw 6 circles. Is it better to define a different VAO and VBO for each circle, or do I have to do that? 2: I want to put an image for each circle. How should I deal with the textures? I mean, for each texture, I define a special uniform.  It does, right?  And I repeat all the steps of defining and creating texture 6 times as if I were placing it on one circle?  I did that, but the images only appeared on one circle

**Comment by Red1C3** [comment]
1: VAOs and VBOs are used for storing data in the VRAM, storing a single circle in a single VAO/VBO is the ideal case since they share the same geometry and hence the since data
2: Textures have equivalent uniforms in the shaders, let's say you only want to use a texture for coloring, you'll define only a single uniform variable in the shader and use it for the coloring, by default, the OpenGL driver assumes you are working with the first slot so you don't need to do any mapping between the texture slots and the shader uniforms, basically you'll create a texture object for each image you have and bind the one you want to be drawn on the mesh **before calling a draw function**

**Comment by MuhammadSulaiman001** [comment]
Hi @DaniaNouro, I've added new sample ([`One-VAO-Two-Textures.cpp`](https://github.com/MuhammadSulaiman001/opengl-lab/blob/master/OpenGLab/LabDemos/src/samples/One-VAO-Two-Textures.cpp)) to illustrate how you can use one VBO/VAO then bind/attach the desired texture just before you show that texture.
